File: 0705-design-hashset/0705-design-hashset.py
class ListNode:
    def __init__(self, key):
        self.key = key
        self.next = None

class MyHashSet:

    def __init__(self):
        self.set = [ListNode(0) for i in range(10**4)]

    def add(self, key: int) -> None:
        index = self.hashFunction(key)
        cur = self.set[index] #head of linked list for index in hashset

        while cur.next: #cur.next != NULL
            if cur.next.key == key:
                return
            cur = cur.next
        cur.next = ListNode(key)
        

    def remove(self, key: int) -> None:
        index = self.hashFunction(key)
        cur = self.set[index] #head of linked list for index in hashset

        while cur.next: #cur.next != NULL
            if cur.next.key == key:
                cur.next = cur.next.next
                return
            cur = cur.next

    def contains(self, key: int) -> bool:
        index = self.hashFunction(key)
        cur = self.set[index] #head of linked list for index in hashset

        while cur.next: #cur.next != NULL
            if cur.next.key == key:
                return True
            cur = cur.next
        return False

    def hashFunction(self, key: int) -> int:
        return key%len(self.set)
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: 0705-design-hashset/0705-design-hashset.cpp
class MyHashSet {
private:
    vector<int> hashset;
public:
    MyHashSet() {
        hashset = {};
    }
    
    void add(int key) {
        if(!contains(key)){
            hashset.push_back(key);
        }
    }
    
    void remove(int key) {
        if(contains(key)){
            hashset.erase(std::remove(hashset.begin(), hashset.end(), key), hashset.end());
        }
    }
    
    bool contains(int key) {
        auto it = find(hashset.begin(), hashset.end(), key);
        if(it == hashset.end()){
            return false;
        }
        return true;
    }

};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */